transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -872,12 +893,94 @@ private AbstractWriteTask(Recycler.Handle handle) {
             super(handle);
         }
 
+        private static int estimateSize(Object o, Map<Class<?>, Integer> classSizes) {
+            if (o == null) {
+                return 8;
+            }
+
+            int answer = 8 + estimateSize(o.getClass(), classSizes, null);
+
+            if (o instanceof ByteBuf) {
+                answer += ((ByteBuf) o).readableBytes();
+            } else if (o instanceof ByteBufHolder) {
+                answer += ((ByteBufHolder) o).content().readableBytes();
+            } else if (o instanceof FileRegion) {
+                answer += 0;
+            } else if (o instanceof byte[]) {
+                answer += ((byte[]) o).length;
+            } else if (o instanceof ByteBuffer) {
+                answer += ((ByteBuffer) o).remaining();
+            } else if (o instanceof CharSequence) {
+                answer += ((CharSequence) o).length() << 1;
+            } else if (o instanceof Iterable<?>) {
+                for (Object m : (Iterable<?>) o) {
+                    answer += estimateSize(m, classSizes);
+                }
+            }
+
+            return align(answer);
+        }
+
+        private static int estimateSize(Class<?> clazz, Map<Class<?>, Integer> classSizes,
+                                        Set<Class<?>> visitedClasses) {
+            Integer objectSize = classSizes.get(clazz);
+            if (objectSize != null) {
+                return objectSize;
+            }
+
+            if (visitedClasses != null) {
+                if (visitedClasses.contains(clazz)) {
+                    return 0;
+                }
+            } else {
+                visitedClasses = new HashSet<Class<?>>();
+            }
+
+            visitedClasses.add(clazz);
+
+            int answer = 8; // Basic overhead.
+            for (Class<?> c = clazz; c != null; c = c.getSuperclass()) {
+                Field[] fields = c.getDeclaredFields();
+                for (Field f : fields) {
+                    if ((f.getModifiers() & Modifier.STATIC) != 0) {
+                        // Ignore static fields.
+                        continue;
+                    }
+
+                    answer += estimateSize(f.getType(), classSizes, visitedClasses);
```

Comment:
@nmittler I think this is handled above already https://github.com/netty/netty/pull/4380#discussion-diff-42629724R917 ?